reactor-core/src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -2100,6 +2101,32 @@ public static <T, D> Flux<T> using(Callable<? extends D> resourceSupplier, Funct
 		return using(resourceSupplier, sourceSupplier, resourceCleanup, true);
 	}
 
+	/**
+	 * Uses a {@link AutoCloseable} resource, generated by a supplier for each individual Subscriber,
+	 * while streaming the values from a Publisher derived from the same resource and makes sure
+	 * the resource is released if the sequence terminates or the Subscriber cancels.
+	 * <p>
+	 * Eager {@link AutoCloseable} resource cleanup happens just before the source termination and exceptions raised
+	 * by the cleanup Consumer may override the terminal event.
+	 * <p>
+	 * <img class="marble" src="doc-files/marbles/usingForFlux.svg" alt="">
+	 * <p>
+	 * For an asynchronous version of the cleanup, with distinct path for onComplete, onError
+	 * and cancel terminations, see {@link #usingWhen(Publisher, Function, Function, BiFunction, Function)}.
+	 *
+	 * @param resourceSupplier a {@link Callable} that is called on subscribe to generate the resource
+	 * @param sourceSupplier a factory to derive a {@link Publisher} from the supplied resource
+	 * @param <T> emitted type
+	 *
+	 * @return a new {@link Flux} built around a disposable resource
+	 * @see #usingWhen(Publisher, Function, Function, BiFunction, Function)
+	 * @see #usingWhen(Publisher, Function, Function)
+	 */
+	public static <T> Flux<T> using(Callable<? extends AutoCloseable> resourceSupplier,
+									Function<? super AutoCloseable, ? extends Publisher<? extends T>> sourceSupplier) {
```

Comment:
This feature caught my eye. For better ease of use, you probably want this signature to be:
```java
public static <T, D extends AutoCloseable> using(Callable<D> resourceSupplier, Function<? super D, ? extends Publisher<? extends T>> sourceSupplier)
```

Reply:
✅ updated, good suggestion!